cylc/flow/scheduler.py

```diff
@@ -744,25 +758,39 @@ def restart_remote_init(self):
             ):
                 distinct_install_target_platforms.append(itask.platform)
 
-        incomplete_init = False
         for platform in distinct_install_target_platforms:
+            # skip remote init for localhost
+            install_target = platform['install target']
+            if install_target == get_localhost_install_target():
+                continue
+            # set off remote init
             self.task_job_mgr.task_remote_mgr.remote_init(
                 platform, self.server.curve_auth,
                 self.server.client_pub_key_dir)
+            # Remote init/file-install is done via process pool
+            self.proc_pool.process()
+            # add platform to map (to be picked up on main loop)
+            self.incomplete_ri_map[install_target] = platform
+
+    def manage_remote_init(self):
+        """Manage the remote init/file install process for restarts.
+            Called within the main loop.
+            Starts file installation when Remote init is complete.
+            Removes complete installations or installations encountering SSH
+            error (remote init will take place on next job submission).
+        """
+        for install_target, platform in self.incomplete_ri_map.copy().items():
```

Comment:
Copying the dict works fine, but we tend to copy the list of items which should be a little quicker:

```suggestion
        for install_target, platform in list(self.incomplete_ri_map.items()):
```

Reply:
Ah great, that makes sense.